components/org.wso2.carbon.identity.api.idpmgt.endpoint/src/main/java/org/wso2/carbon/identity/api/idpmgt/endpoint/impl/IdpsApiServiceImpl.java

```diff
@@ -0,0 +1,329 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.carbon.identity.api.idpmgt.endpoint.impl;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.api.idpmgt.IDPMgtBridgeService;
+import org.wso2.carbon.identity.api.idpmgt.IDPMgtBridgeServiceClientException;
+import org.wso2.carbon.identity.api.idpmgt.IDPMgtBridgeServiceException;
+import org.wso2.carbon.identity.api.idpmgt.IdPUtils;
+import org.wso2.carbon.identity.api.idpmgt.endpoint.EndpointUtils;
+import org.wso2.carbon.identity.api.idpmgt.endpoint.IdpsApiService;
+import org.wso2.carbon.identity.api.idpmgt.endpoint.dto.ClaimConfigDTO;
+import org.wso2.carbon.identity.api.idpmgt.endpoint.dto.FederatedAuthenticatorConfigDTO;
+import org.wso2.carbon.identity.api.idpmgt.endpoint.dto.IdPDetailDTO;
+import org.wso2.carbon.identity.api.idpmgt.endpoint.dto.JustInTimeProvisioningConfigDTO;
+import org.wso2.carbon.identity.api.idpmgt.endpoint.dto.PermissionsAndRoleConfigDTO;
+import org.wso2.carbon.identity.api.idpmgt.endpoint.dto.ProvisioningConnectorConfigDTO;
+import org.wso2.carbon.identity.application.common.model.ClaimConfig;
+import org.wso2.carbon.identity.application.common.model.FederatedAuthenticatorConfig;
+import org.wso2.carbon.identity.application.common.model.IdentityProvider;
+import org.wso2.carbon.identity.application.common.model.JustInTimeProvisioningConfig;
+import org.wso2.carbon.identity.application.common.model.PermissionsAndRoleConfig;
+import org.wso2.carbon.identity.application.common.model.ProvisioningConnectorConfig;
+
+import java.util.List;
+import javax.ws.rs.core.Response;
+
```

Comment:
Better to add some class comments even though this is auto generated since we are modifying this manually.